internal/service/encryptionatrest/resource_encryption_at_rest.go

```diff
@@ -376,9 +371,15 @@ func (r *encryptionAtRestRS) Delete(ctx context.Context, req resource.DeleteRequ
 		return
 	}
 
-	conn := r.Client.Atlas
+	enabled := false
+	connV2 := r.Client.AtlasV2
 	projectID := encryptionAtRestState.ProjectID.ValueString()
-	_, err := conn.EncryptionsAtRest.Delete(ctx, projectID)
```

Comment:
Delete operation was actually a PATCH operation doing a soft delete, that's why it's now an Update in the new SDK where we set everything as disabled (enabled: false) [api docs](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Encryption-at-Rest-using-Customer-Key-Management/operation/updateEncryptionAtRest)